go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -139,10 +140,11 @@ def run_go_mod_tidy
           command = "go mod tidy -e"
 
           # we explicitly don't raise an error for 'go mod tidy' and silently
-          # continue here. `go mod tidy` shouldn't block updating versions
-          # because there are some edge cases where it's OK to fail (such as
-          # generated files not available yet to us).
-          Open3.capture3(environment, command)
+          # continue with an info log here. `go mod tidy` shouldn't block
+          # updating versions because there are some edge cases where it's OK to fail
+          # (such as generated files not available yet to us).
+          _, stderr, status = Open3.capture3(environment, command)
+          Dependabot.logger.info "Failed to `go mod tidy`: #{stderr}" unless status.success?
```

Comment:
Unfortunately, we _probably_ can't log the `stderr` output for fear it might contain creds or leak PII data into our logs. I say _probably_ because I'm not 100% sure, give me a day or two to double-check internally as we are currently working through the same question for:
* https://github.com/dependabot/dependabot-core/pull/6521

iff we can't do this, then we'll need to do something like this... but again, wait to hear from me before changing it:
```suggestion
          Dependabot.logger.info "Unexpected error while running `go mod tidy`, continuing without tidying" unless status.success?
```

Reply:
IIUC the above message will not help much in debugging as `go mod tidy` can fail for a variety of reasons. Let's consider another way if it's sensitive to reveal.

Reply:
The fact that it failed at all is actually very useful, as historically the problem has typically been something in the underlying repo, not the changes that Dependabot made. So the user can simply try running `go mod tidy` locally against their repo and see what it spits out. But do agree that showing the full error would be even more useful. Give me a day or two to sync up internally on this.

Reply:
So the upshot was that it's fine to expose the error in the job logs. 

The concern that hasn't been resolved internally revolves around raising exceptions since they get propagated to a different system, with a different set of requirements on what can/can't be stored. But since this only logs and doesn't raise any error, it won't matter.

Reply:
I am curious to hear why we picked `info` here when this is an error?

Reply:
Because it can happen when a user does nothing wrong.

Reply:
That does not justify the level, in my opinion, as it is still an error

Reply:
I guess I don't understand your argument, I'd recommend opening a new issue or submitting a PR directly. Note that commenting on a merged PR isn't a good practice as it easily gets stalled.